### ISSUE TITLE
Type Variables in Native Toplevel

### ISSUE THREAD

**Issue by NathanReb** [NathanReb/ocaml-native-toplevel-experience#1]
The native toplevel does not retain type var names as the bytecode toplevel does. When a type with type parameters is defined in a native toplevel session, with those parameters explicitly named, future evaluation may not retain this name and use the regular `'a`, `'b`, etc. instead.

Here's an example, taken from [ocaml-jit's test suite](https://github.com/NathanReb/ocaml-jit/blob/main/tests/preserve_type_vars.ml):
```ocaml
# module A : sig 
    type ('foo, 'bar) t

    val get_foo : ('foo, _) t -> 'foo option
  end = struct
    type ('foo, 'bar) t =
      | Foo of 'foo
      | Bar of 'bar

    let get_foo = function
      | Foo foo -> Some foo
      | Bar _ -> None
  end
  ;;
module A :
  sig type ('foo, 'bar) t val get_foo : ('foo, 'a) t -> 'foo option end
# A.get_foo;;
- : ('foo, 'a) A.t -> 'foo option = <fun>
```
This is what you get in a bytecode toplevel, the `'foo` type variable is preserved in the toplevel evaluation of `A.get_foo`. Running the same phrases in a native toplevel would get you:
```
- : ('a, 'b) A.t -> 'a option = <fun>
```

This has been fixed in ocaml-flambda/flambda-backend#188.

The upstream patch is awaiting review here ocaml/ocaml#10712.

**Comment by dra27** [comment]
This is fixed in 4.14!

**Comment by NathanReb** [comment]
Wonderful, thanks for taking care of this!
